juju/juju
# relation status implemented as struct rather than relation field

## Description of change

Relation status was being modelled as a field in the state relation doc.
We now model it like for other entities as an entry in the statuses collection.

Most of the changes are mechanical, but the state relation status watcher needed to be rewritten to account for needed to get its data from 2 collections.

The juju/description  repo has been modified to model status the new way and the dep for this pr is updated. The juju/decription is https://github.com/juju/description/pull/22

The second commit, https://github.com/juju/juju/pull/7831/commits/24bdeb80b5c2ec359a9177fda4433157d211e083, fixes an issue that came up in testing. When the uniter is initialising, it needs to treat any suspended relations as still active, not just those in scope. Otherwise it attempts to incorrectly remove the unit state dir.

## QA steps

Run up a cmr scenario.
Set status for relation on offering side to suspended.
Ensure relation on consuming side is broken.
Set relation to resumed. See that consuming relation recovers.
Delete relation on offering side. 
Relation on consuming side is removed.
